tools/webdriver/webdriver/bidi/modules/browsing_context.py

```diff
@@ -178,5 +179,8 @@ def _print(self, result: Mapping[str, Any]) -> Any:
         return result["data"]
 
     @command
-    def set_viewport(self, context: str, viewport: Optional[Mapping[str, Any]] = None) -> Mapping[str, Any]:
-        return {"context": context, "viewport": viewport}
+    def set_viewport(self, context: str, viewport: Union[Optional[Mapping[str, Any]], str] = None) -> Mapping[str, Any]:
+        params = {"context": context}
+        if viewport != 'undefined':
```

Comment:
@whimboo PTAL at the problem with the undefined vs null. I am okay with using the 'undefined' as special keyword. Let us know what you think.

Reply:
There were discussions around this at https://github.com/web-platform-tests/wpt/issues/34697 .

In think at the moment you can rely on the raw `send_blocking_command` fixture to send specific payloads. That's what we already do to test maxDomDepth = null for script evaluate. Granted this is the opposite scenario, because `maxDomDepth` is optional and would be omitted if =None, but I think the same strategy would work.